### ISSUE TITLE
DataViews: Style destructive secondary actions

### ISSUE THREAD

**Issue by doekenorg** [WordPress/gutenberg#66810]
## What problem does this address?
Actions can be made "destructive" by setting the `isDestructive` attribute to `true`. When viewing this as a primary action, this is apparent by the red color. These styles are not applied to a secondary destructive action.

## What is your proposed solution?
Apply destructive styling on secondary actions, and provide a `is-destructive` class there as well, for any required customisation.

![Image](https://github.com/user-attachments/assets/d522e0c4-1051-4d4e-b78a-6d914f19e3d7)

CC: @WordPress/gutenberg-components

**Comment by himanshupathak95** [comment]
Hey @doekenorg, Thanks for bringing this up!

I think this would be a great accessibility enhancement and I would like to help implement this feature.

Could you please help me with:
1. Steps to reproduce the current behavior
2. The WordPress and plugin versions you're using

Currently, I'm not able to replicate this exact behavior and having these details would help me get started.

Thanks in advance!

**Comment by doekenorg** [comment]
Hi @himanshupathak95,

You can reproduce this behaviour on the latest version. If you install the storybook locally, you can reproduce it on there. 

After reading https://developer.wordpress.org/block-editor/contributors/code/getting-started-with-code-contribution/ you should be able to set it up locally. If you then run `npm install && npm run storybook:dev` you can use the https://github.com/WordPress/gutenberg/blob/trunk/packages/dataviews/src/components/dataviews/stories/index.story.tsx#L23 file to make changes and try it out.

**Comment by ntsekouras** [comment]
I think we had that at first and later removed.. 🤔 

--cc @WordPress/gutenberg-design

**Comment by jameskoster** [comment]
I'd encourage us to hold on any changes here, and await the outcome of https://github.com/WordPress/gutenberg/issues/58390 which seeks to define when to use destructive styling at a higher level.

**Comment by youknowriad** [comment]
What's the path forward here, we have two issues about destructive styling but unclear actionable items.

**Comment by jameskoster** [comment]
Yes ideally we align on the logic and document it somewhere for future reference. I saw some designs from @poligilad-auto, maybe they can share here or in https://github.com/WordPress/gutenberg/issues/58390.

**Comment by poligilad-auto** [comment]
Thanks for the ping @jameskoster. 

Here’s the proposed logic:

- Destructive buttons don’t use a destructive color; they use the regular button color.
- We always show a modal after clicking on a destructive button.

In the modal:

- We clearly explain the implications of the destructive action.
- If the action is irreversible the main button uses a destructive color.
- If the action is reversible the main button uses the regular button color.

Here are some images to illustrate.

<img width="1104" height="746" alt="Image" src="https://github.com/user-attachments/assets/27e4f163-2e84-461b-99f8-337f575797a4" />

<img width="1104" height="746" alt="Image" src="https://github.com/user-attachments/assets/eb708095-5dd5-4283-87aa-0a2c62487e3e" />

**Comment by jasmussen** [comment]
An item for the wishlist (no blockers): a nicer shade of red 😊 🙏

**Comment by jameskoster** [comment]
@poligilad-auto This looks good to me, but I wonder how you see deleting a block fitting into this convention. Does that count as 'destructive'? It would be annoying to see a modal in that flow I think. How about deleting a custom shadow or font size preset in global styles? We might need to nail down what 'destructive' means more explicitly.
